src/Microsoft.AspNetCore.Mvc.Core/Internal/ParameterDefaultValues.cs

```diff
@@ -21,34 +21,38 @@ public static object[] GetParameterDefaultValues(MethodInfo methodInfo)
 
             for (var i = 0; i < parameters.Length; i++)
             {
-                var parameterInfo = parameters[i];
-                object defaultValue;
+                values[i] = GetParameterDefaultValue(parameters[i], considerDefaultValueAttribute: true);
+            }
+
+            return values;
+        }
 
-                if (parameterInfo.HasDefaultValue)
+        public static object GetParameterDefaultValue(ParameterInfo parameterInfo, bool considerDefaultValueAttribute)
+        {
+            // In a scenario where the method signature is like "void Foo(Guid id = default(Guid))", 
+            // HasDefaultValue returns true, but the DefaultValue is null. This happens only in case of Reflection,
+            // whereas during execution time, calling the method "Foo()" does indeed supply the default value of
+            // 00000000-0000-0000-0000-000000000000 to the parameter "id".
+            if (parameterInfo.HasDefaultValue && parameterInfo.DefaultValue != null)
```

Comment:
@pakrym found some shennanigans around default values when he was doing DI stuff. See https://github.com/aspnet/DependencyInjection/blob/dev/shared/Microsoft.Extensions.ParameterDefaultValue.Sources/ParameterDefaultValue.cs. We might need some of those here.

Reply:
Yep, please use that thing.